module/zfs/zfs_znode.c

```diff
@@ -915,7 +915,11 @@ zfs_mknode(znode_t *dzp, vattr_t *vap, dmu_tx_t *tx, cred_t *cr,
 	VERIFY(sa_replace_all_by_template(sa_hdl, sa_attrs, cnt, tx) == 0);
 
 	if (!(flag & IS_ROOT_NODE)) {
-		*zpp = zfs_znode_alloc(zfsvfs, db, 0, obj_type, obj, sa_hdl);
+		do {
+			*zpp = zfs_znode_alloc(zfsvfs, db, 0, obj_type, obj,
```

Comment:
Right.  Our `kmem_cache_alloc()` can't fail but `security_inode_alloc()` can.  I'm not thrilled with retrying in here, but this should be exceedingly rare and it's better than a VERIFY.  I like the idea of adding a comment clearly describing the callpath we're worried about, I'll add a description of the issue when I refresh this.  Thanks for taking the time to look it over!